client/src/app/auth/login.tsx

```diff
@@ -1,7 +1,52 @@
+import { ChangeEvent, FormEvent, useState } from "react";
+import Button from "../components/button";
+import TextField from "../components/text-field";
+
 export default function Login() {
   return (
     <div>
       <h1>Log in</h1>
+      <ExecLogIn />
     </div>
   );
 }
+
+
+function ExecLogIn() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleInputChange(e: ChangeEvent<HTMLInputElement>): void {
+    const { name, value } = e.target;
+    if (name === 'username') {
+      setUsername(value);
+    } else if (name === 'password') {
+      setPassword(value);
+    }
+  }
+
+  function handleLogin(e: FormEvent<Element>): void {
+    throw new Error("Function not implemented.");
```

Comment:
Add `e.preventDefault()` here so the page doesn't refresh on button click. 

Also, you can just log to the console or have an alert instead of throwing an error for now since this error isn't caught by anything.